swap_meet/vendor.py

```diff
@@ -1,2 +1,207 @@
+    output: 
+        True - if self and othr_vendor have items that the other wants to swap
+            Side effect- the inventory parameter for self and other_vendor is modified by having the new item and
+                removing the old item
+        False - if self or other_vendor does not have its corrsponding item in their inventory parameter
+        '''
+
+        #Check each of the vendors has the correct item to swap 
+
+        if my_item in self.inventory and others_item in other_vendor.inventory:
+            
+            self.inventory.remove(my_item)
+            other_vendor.inventory.remove(others_item)
+            
+            self.inventory.append(others_item)
+            other_vendor.inventory.append(my_item)
+            
+            return True 
+        
+        # Return False if one of the vendors does not have the item
+        return False
+
+
+    
+    def swap_first_item(self, other_vendor):
+        '''
+        input:
+            self
+            other_vendor - another object of the same class (vendor)
+        output: 
+            False - If either of the vendors have empty inventories 
+            True - If both of the vendors have at least one item in their inventories 
+                and a swap is done 
+        '''
+
+        if len(self.inventory) and len(other_vendor.inventory):
+
+            my_item = self.inventory[0]
+            their_item = other_vendor.inventory[0]
+            self.swap_items(other_vendor,my_item,their_item)
+
+            return True 
+
+        return False 
+    
+    
+
+    def get_by_category(self, category):
+        '''
+        input:
+            self -
+            category - str specifiying which type of item the inventory of self is going to be filtered by 
+        output: 
+            items_in_category - list of items that self has in its inventory under the category
+                specified as an argument
+        '''
+
+        items_in_category = []
+
+        for item in self.inventory:
+
+            if item.get_category() == category:
+                items_in_category.append(item)
+
+        return items_in_category
+    
+
+    
+
+    def get_best_by_category(self, category):
+
+        '''
+        input:
+            self
+            category - str specific category of items that inventory will be filtered by 
+        output:
+            best_item - will return the item with the best rating in the specified category
+            None - if there is no item with that specific category 
+        '''
+
+        # List of items in selfs inventory that have the specified category
+
+        items_in_category = self.get_by_category(category)
+
+        if len(items_in_category) == 0:
+            return None
+        
+        best_item = items_in_category[0]
+
+        for item in items_in_category:
+            if item.condition > best_item.condition:
+                best_item = item
+        return best_item
```

Comment:
Great job with this. It's good to have the practice doing this with loops!

You can also do this with max, once you're comfortable the loop-y way:

```python
return max(items_in_category, key=lambda item: item.condition)
```